clconf/template.go

```diff
@@ -125,17 +127,27 @@ func processTemplate(paths pathWithRelative, dest string, value interface{},
 		return result, fmt.Errorf("Error processing template: %v", err)
 	}
 
-	err = ioutil.WriteFile(target, []byte(content), mode)
-	if err != nil {
-		return result, err
-	}
-
-	if !options.KeepExistingPerms {
-		// WriteFile only uses the mode if the file is created during write
-		err = os.Chmod(target, mode)
+	if options.KeepEmpty || content != "" {
+		// This won't change the mode if it already exists
+		err = ioutil.WriteFile(target, []byte(content), mode)
 		if err != nil {
 			return result, err
 		}
+
+		if !options.KeepExistingPerms {
+			// WriteFile only uses the mode if the file is created during write
+			// and even then filters on the umask so we os.Chmod to get the
+			// requested perms.
+			err = os.Chmod(target, mode)
+			if err != nil {
+				return result, err
+			}
+		}
+	} else { //Don't keep the empty result
+		_, err = os.Stat(target)
+		if err == nil {
+			os.Remove(target)
```

Comment:
Dont wanna check this error?  Should clconf template processing report a failure if it leaves the file in place?

Reply:
[Fixed](https://github.com/pastdev/clconf/pull/13/commits/e86bd2cdf541c247e3007ed37169507d2d56a096)